### PR TITLE
Documented CALICO_NAT_OUTGOING Parameter

### LINKED ISSUE #516: Provide an environment variable to control `nat-outgoing`
Currently in v1.0.2, the default value for `nat-outgoing` is false. To enable out going connection for the containers, I have to follow these steps: http://docs.projectcalico.org/master/usage/external-connectivity. Also I was informed that with the next release the default value for `nat-outgoing` would be false. This is great but it would be really useful if this option was configurable via an environment variable. (Something similar to CALICO_IPIP_ENABLED, say CALICO_NAT_OUTGOING?)

### PR DESCRIPTION
**Pull request by VincentS**
Added documentation for CALICO_NAT_OUTGOING  added by pull request [#1640](https://github.com/projectcalico/calicoctl/pull/1640).

Fixes #516

### CODE REVIEW

**Comment by tmjd** [comment]
This looks good but will need some updates after https://github.com/projectcalico/calicoctl/pull/1640 is finalized.

File: master/reference/node/configuration.md [hunk 1]
```diff
@@ -27,6 +27,8 @@ The `calico/node` container is primarily configured through environment variable
 | CALICO_IPV4POOL_CIDR | The IPv4 Pool to create if none exists at start up. It is invalid to define this variable and NO_DEFAULT_POOLS. | IPv4 CIDR | 192.168.0.0/16 |
 | CALICO_IPV6POOL_CIDR | The IPv6 Pool to create if none exists at start up. It is invalid to define this variable and NO_DEFAULT_POOLS. | IPv6 CIDR | fd80:24e2:f998:72d6::/64 |
 | CALICO_IPV4POOL_IPIP | IPIP Mode to use for the IPv4 POOL created at start up. | off, always, cross-subnet | off |
+| CALICO_IPV4POOL_NAT_OUTGOING | Controls NAT Outgoing `nat-outgoing` for the IPv4 IPPOOL created at start up. | boolean | true |
```
Comment (tmjd) [inline master/reference/node/configuration.md]: We haven't referenced the specific flag in the Resource in other descriptions so I think you should remove the `nat-outgoing` from both of these entries.
nit: `IPv4 IPPOOL` should be `IPv4 Pool`

File: master/reference/node/configuration.md [hunk 2]
```diff
@@ -27,6 +27,8 @@ The `calico/node` container is primarily configured through environment variable
 | CALICO_IPV4POOL_CIDR | The IPv4 Pool to create if none exists at start up. It is invalid to define this variable and NO_DEFAULT_POOLS. | IPv4 CIDR | 192.168.0.0/16 |
 | CALICO_IPV6POOL_CIDR | The IPv6 Pool to create if none exists at start up. It is invalid to define this variable and NO_DEFAULT_POOLS. | IPv6 CIDR | fd80:24e2:f998:72d6::/64 |
 | CALICO_IPV4POOL_IPIP | IPIP Mode to use for the IPv4 POOL created at start up. | off, always, cross-subnet | off |
```
Comment (tmjd) [inline master/reference/node/configuration.md]: While you're in here would you change `POOL` to `Pool`.  Thanks

File: master/reference/node/configuration.md [hunk 3]
```diff
@@ -27,6 +27,8 @@ The `calico/node` container is primarily configured through environment variable
...
+| CALICO_IPV4POOL_NAT_OUTGOING | Controls NAT Outgoing `nat-outgoing` for the IPv4 IPPOOL created at start up. | boolean | true |
+| CALICO_IPV6POOL_NAT_OUTGOING | Controls NAT Outgoing `nat-outgoing` for the IPv6 IPPOOL created at start up. | boolean | false |
```
Comment (tmjd) [inline master/reference/node/configuration.md]: same as above

**Comment by tmjd** [comment]
@VincentS Could you please rebase this and get rid of that merge commit from me?  I hit the appealing "Update branch" button and it created that merge commit, sorry I'd never used it before and need to remember to never use it again.
After the rebase then I think this is good.

**Comment by VincentS** [comment]
Done!

**Comment by tmjd** [comment]
LGTM
@robbrockbank What do you think?

**Comment by robbrockbank** [comment]
LGTM

**Pull request merged**